src/queryExecutionContext/EndpointComponent/UnorderedDistinctEndpointComponent.ts

```diff
@@ -0,0 +1,33 @@
+import { sha1 } from "crypto-hash";
+import stableStringify from "fast-json-stable-stringify";
+import { Response } from "../../request";
+import { ExecutionContext } from "../ExecutionContext";
+
+/** @hidden */
+export class UnorderedDistinctEndpointComponent implements ExecutionContext {
```

Comment:
Can we make these classes implement some `DistinctEndpointComponentBase` class and make these private nested. The base class can then have a factory create method that take the `DistinctType` and `new`s up the appropriate ExecutionContext.

Reply:
Same as other comments, prefer to keep this explicit for now. Trying to come back and refactor the query pipeline later and would like to have less inheritance and complexity